Updated things

resynth1943 opened the pull request:
I've updated metadata, added an `npmignore` and a `gitignore`, cleaned up the `package.json`, and added a programmatic API, e.g

```js
const { createPaste } = require('hastebin');
createPaste('abc').then(processPasteUrl);
```

so you can use Hastebin in node without "shelling out".

Oh, and I've updated the README and tests to match. Have I missed anything?

Commit:
Add a Node gitignore

I copied the .gitignore file from
https://github.com/github/gitignore/blob/master/Node.gitignore

Commit:
Add an npmignore

Commit:
Add the package-lock file

Commit:
Improve the package.json

Commit:
Move CLI to src, remove /bin file

Commit:
Create JS api

Commit:
Add testsf for JS api

Commit:
Add typings

Commit:
Update README

Commit:
Avoid using es6 syntax

rahatarmanahmed commented on README.md:
````diff
@@ -23,16 +28,31 @@ Examples:
+```js
+var hastebin = require('hastebin')
+
+hastebin.createPaste('url', {
````
Can you change the example text from `'url'` to something more... text like? Took me a second to figure out that was the text content of the paste.

resynth1943 replied:
Oh, sorry about that. I was inbetween two different things and forgot that that's the parameter for the *content*.

resynth1943 replied:
Boom :D

resynth1943 replied:
Pushed :+1:

rahatarmanahmed commented on test/test.js:
```diff
@@ -1,21 +1,27 @@
-.then(function (stdin) {
-  var hastebinUrl = stdin.trim()
-  assert(hastebinUrl.match(/^https:\/\/hastebin.com\/.*/) != null, 'stout must be a hastebin url')
+function testOutput (input) {
```
Can you rename this to something more descriptive like `assertHastebinUrlMatchesTestData` or `assertHastebinUrlMatchesText`

resynth1943 replied:
Done

rahatarmanahmed commented on src/cli.js:
```diff
@@ -0,0 +1,37 @@
+  } else {
+    console.log(url.resolve(hasteServer, result.body.key))
+  }
+})
```
Why not reuse the programmatic API?

resynth1943 replied:
I can't argue with that, lol

resynth1943 replied:
Consider it done :)

rahatarmanahmed reviewed (commented):
A really solid contribution, thanks!

rahatarmanahmed commented:
Also, if you want to use es6 you're welcome to bump the engine requirement to node 6. But keeping node 4 support is okay too 👍

Commit:
Add get-stream dependency

Commit:
Clean up CLI

Commit:
Give the assertion function a better name in test src

Give test function a better name

Give the test function a more descriptive name

resynth1943 commented:
@rahatarmanahmed I don't see the point of bumping the minimum engine version requirement up to v6 just for fancy syntax, to be honest. Thanks, though :)

Commit:
Fix example in README

resynth1943 commented:
By the way, would you be open (pardon the pun) to me adding a flag that automatically opens the link in the browser, via a module like `open`?

e.g

```sh
echo "abc" | hastebin --browser
# or 
echo | "abc" | hastebin -b (or -B?)
```

This would mitigate having to click the link / copy it into the omnibar. It's lazy, I know :D

rahatarmanahmed commented:
I'm not opposed to it but personally I think functionality like that is best implemented by composing other unix tools. Piping into `xclip` or `open` is free ;)

rahatarmanahmed commented:
Published in v0.1.0!

resynth1943 commented:
Yay! :D
